src/server/routes.ts

```diff
@@ -1,12 +1,182 @@
-import express from "express";
+import express, { json, urlencoded } from "express";
 import cors from "cors";
+import Controller from "./controller/index";
+import { Trade } from "./types/index";
+import Gateway from "./gateways/gateway";
+
+const envPort: string | undefined = process.env.PORT;
+const controller = new Controller();
+class ThinGateway extends Gateway {}
+const gateway = new ThinGateway();
 
 const app = express();
 
+app.use(urlencoded());
+app.use(json());
 app.use(
   cors({
     origin: "*",
   })
 );
 
 app.use(express.static(`${__dirname}/../client/dist`));
+
+app.put("/trade/:user/write", async function (req, res) {
+  const body: Trade = req.body;
+  let precedence: boolean = false;
+
+  if (!req.body.id) {
+    res.sendStatus(400);
+    return;
+  }
+
+  try {
+    const result: {
+      rows: Trade[];
+    } = await controller.read(body.id);
+
+    result.rows.length > 0 ? (precedence = true) : (precedence = false);
+  } catch (err) {
+    gateway.logger.log("error", "Error:", err);
+
+    // Escape early if read fails
+    res.sendStatus(500);
+    return;
+  }
+
+  if (!!precedence) {
+    body.created_at = controller.generateDate();
+
+    try {
+      const result: {
+        rows: {
+          id: string;
+        }[];
+      } = await controller.update(body);
+
+      res.status(200).send(result);
+      return;
+    } catch (err) {
+      gateway.logger.log("error", "Error:", err);
+      res.sendStatus(500);
```

Comment:
**Comment**: We should also be sending a message instead of a flat 500, something client-worthy, i.e. "Internal Server Error" (the most common use case).

Reply:
Shouldn't the method sendStatus send in the message as well? Or would you prefer I do the traditional way of res.status(500).send("Internal Server Error")?

> res.sendStatus(500); // equivalent to res.status(500).send('Internal Server Error')

Reply:
Updated res.sendStatus to res.status(500).send("Internal Server Error").